### ISSUE TITLE
Can you share the parameters of GAN model?

### ISSUE THREAD

**Issue by Harxis** [ml-postech/gradient-inversion-generative-image-prior#1]
Retrain the GAN model is time wasting, can you please share the pretrained GAN model?

**Comment by jc5201** [comment]
We included a link of pretrained model and updated readme. 
But we are still not sure about the license.
Please let me know if you have any idea about the license of pretrained models.

===

For stylegan, we did not included pretrained models. You should bring your model and edit `inversefed/porting.py`. For stylegan2 model and FFHQ dataset, you can download [this model](https://www.dropbox.com/s/k5ql7yyjl2i9dno/Gs.pth?dl=0) into `inversefed/genmodels/stylegan2/Gs.pth`. Note that the model is obtained from [other stylegan repo](https://github.com/adriansahlman/stylegan2_pytorch) and follows NVIDIA license. That is, it is restricted to non-commercial use only.

**Comment by Harxis** [comment]
Thank you for your reply. I'm asked for the pretrained model for reproducing the results reported by the paper instead of commercial use, so I think it follows NVIDIA license. And I am confused about the performance of stylegan2 model trained by ImageNet dataset. To the best of my knowledge, diversity may decrease the performance of GAN model. And I have tried to train a stylegan2 model by ImageNet, I have found it is hard to converge, could you please share your experience on training such a GAN.

**Comment by jc5201** [comment]
We can share a stylegan2 model for imagenet. https://www.dropbox.com/s/kakbu8nmu7fnr6g/imagenet.pth?dl=0
We experienced the model works well with gan inversion setting (not gradient inversion setting), but it is hard to produce natural images when sampled from random latent vectors. 

However, we recommend you to compare FFHQ results first, because the face images and corresponding generative models have much small diversity. The results will be more strong support for the advantages of GIAS (than that of Imagenet).

**Comment by Harxis** [comment]
Thank you sooooooooo much!

**Comment by Harxis** [comment]
> We can share a stylegan2 model for imagenet. https://www.dropbox.com/s/kakbu8nmu7fnr6g/imagenet.pth?dl=0 We experienced the model works well with gan inversion setting (not gradient inversion setting), but it is hard to produce natural images when sampled from random latent vectors.
> 
> However, we recommend you to compare FFHQ results first, because the face images and corresponding generative models have much small diversity. The results will be more strong support for the advantages of GIAS (than that of Imagenet).

Thank you soooooooooo much!
